### PR TITLE
[SPARK-45007][INFRA] Fix merged pull requests resolution

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This PR fixes the resolution for merged pull requests. It appears that the user "asfgit" is no longer closing pull requests, but rather the committers are.

### Why are the changes needed?

Bugfix, make the merge script re-runnable again if you accidentally abort cherry-pick or change your mind later for backporting.

### Does this PR introduce _any_ user-facing change?

maintainer only

### How was this patch tested?

```
>>> for e in pr_events:
...    print(e["event"], e["commit_id"])
...
labeled None
labeled None
labeled None
review_requested None
review_requested None
head_ref_force_pushed None
head_ref_force_pushed None
head_ref_force_pushed None
head_ref_force_pushed None
closed 76a32d3df0fe8e857221662657936e5ae336d3ee
```

```
./dev/merge_spark_pr.py
git rev-parse --abbrev-ref HEAD
Which pull request would you like to merge? (e.g. 34): 42714
Pull request 42714 has already been merged, assuming you want to backport
['git', 'rev-parse', '--quiet', '--verify', '76a32d3df0fe8e857221662657936e5ae336d3ee^{commit}']
Found commit 76a32d3df0fe8e857221662657936e5ae336d3ee:
.
.
.
Closes #42714 from HyukjinKwon/SPARK-45000.

Authored-by: Hyukjin Kwon <gurwls223@apache.org>
Signed-off-by: Dongjoon Hyun <dhyun@apple.com>
Enter a branch name [branch-3.5]:
```

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @dongjoon-hyun @HyukjinKwon @cloud-fan thanks

**Review by cloud-fan — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]

File: dev/merge_spark_pr.py [hunk 1]
```diff
@@ -611,9 +611,9 @@ def main():
     pr_repo_desc = "%s/%s" % (user_login, base_ref)
 
     # Merged pull requests don't appear as merged in the GitHub API;
-    # Instead, they're closed by asfgit.
+    # Instead, they're closed by committers.
     merge_commits = [
-        e for e in pr_events if e["actor"]["login"] == "asfgit" and e["event"] == "closed"
+        e for e in pr_events if e["event"] == "closed" and e["commit_id"] is not None
     ]
```
Comment (dongjoon-hyun) [inline dev/merge_spark_pr.py:617]: Could you fix Python linter failure by running `dev/reformat-python`, @yaooqinn ? It seems to ask to change like the following.
```python
-    merge_commits = [
-        e for e in pr_events if e["event"] == "closed" and e["commit_id"] is not None
-    ]
+    merge_commits = [e for e in pr_events if e["event"] == "closed" and e["commit_id"] is not None]
```

Comment (yaooqinn) [reply dev/merge_spark_pr.py]: Nice catch! Thanks @dongjoon-hyun

**Comment by yaooqinn** [comment]
Thanks @dongjoon-hyun @cloud-fan and @HyukjinKwon for the check

**Pull request closed without merging**